activerecord/test/cases/calculations_test.rb

```diff
@@ -1242,6 +1242,11 @@ def test_calculation_with_query_cache
       count = ShipPart.count
       assert_async_equal count, ShipPart.async_count
     end
+
+    ShipPart.cache do
+      count = ShipPart.async_count
+      assert_equal count.value, ShipPart.async_count.value
```

Comment:
For consistency?

```suggestion
      assert_async_equal count.value, ShipPart.async_count
```

Reply:
Will not work, because `nil?` method is not defined on the `ActiveRecord::Promise`.

Reply:
Huh? If `count.value` returns a promise, something is wrong anyway, right? (Which is the bug we're talking about here.)

Reply:
Yeah, thanks. I missed that the `count.value` stayed in your example, so it works.